src/app/Library/CrudPanel/Traits/FieldsProtectedMethods.php

```diff
@@ -69,7 +68,7 @@ public function makeSureFieldHasMultiple($field)
      */
     public function overwriteFieldNameFromDotNotationToArray($field)
     {
-        if (! is_array($field['name']) && strpos($field['name'], '.') !== false) {
+        if (! is_array($field['name']) && strpos($field['name'], '.') !== false && isset($field['relation_type']) && in_array($field['relation_type'], ['HasOne', 'MorphOne'])) {
```

Comment:
This conditional has gotten to be very long, we can probably return early to be cleaner:

```suggestion
        // if the field name isn't a string with dot notation, nothing to do
        if (is_array($field['name']) || strpos($field['name'], '.') == false) {
            return $field;
        }

        // if no HasOne or MorphOne relationship defined, nothing to do  
        if (!isset($field['relation_type']) || !in_array($field['relation_type'], ['HasOne', 'MorphOne'])) {
            return $field;
        }
        
```

Reply:
Now comes the question - are we SURE we only want to rewrite dot notation to array for `HasOne` and `MorphOne`? Why this added condition? 

What else is there that uses dot notation and SHOULDN'T have it turned into array?